Sources/TerminalAPIKit/EncryptedMesssage/EncryptedMessage.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  EncryptedMessage.swift
+//  TerminalAPIKit
+//
+//  Copyright (c) 2022 Adyen N.V.
+//
+
+import Foundation
+
+/// An encrypted message that can be sent to or received from a Nexo device.
+public final class EncryptedMessage: Codable {
+    
+    /// The header of the message.
+    public let header: MessageHeader
+    
+    /// The encrypted body of the message.
+    public let body: Data
+    
+    /// The security trailer of the message.
+    public let securityTrailer: SecurityTrailer
+    
+    /// Initializes the encrypted message.
+    ///
+    /// - Parameters:
+    ///   - header: The header of the message.
+    ///   - body: The encrypted body of the message.
+    ///   - securityTrailer: The security trailer of the message.
+    public init(header: MessageHeader, body: Data, securityTrailer: SecurityTrailer) {
+        self.header = header
+        self.body = body
+        self.securityTrailer = securityTrailer
+    }
+    
+    /// Initializes the encrypted message by decoding from the given decoder.
+    ///
+    /// This initializer throws an error if reading from the decoder fails, or if the data read is corrupted or otherwise invalid.
+    ///
+    /// - Parameter decoder: The decoder to read data from.
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: StringCodingKey.self)
+        
+        guard let subcontainerKey = container.allKeys.first else {
+            let context = DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "Empty container.")
+            
+            throw DecodingError.dataCorrupted(context)
+        }
+        
+        let subcontainer = try container.nestedContainer(keyedBy: StringCodingKey.self, forKey: subcontainerKey)
+        header = try subcontainer.decode(MessageHeader.self, forKey: "MessageHeader")
+        body = try subcontainer.decode(Data.self, forKey: "NexoBlob")
+        securityTrailer = try subcontainer.decode(SecurityTrailer.self, forKey: "SecurityTrailer")
```

Comment:
why not create a proper `CodingKey` enum instead of hard coded strings?

Reply:
This has to do with some runtime decoding magic created by @joostvandijk . I tried making a usual enum, but decoding does not work in that case